easybuild/tools/systemtools.py

```diff
@@ -137,34 +174,73 @@ def get_total_memory():
     return memtotal
 
 
+def get_cpu_architecture():
+    """
+    Try to detect the CPU architecture
+
+    :return: a value from the CPU_ARCHITECTURES list
+    """
+    aarch32_regex = re.compile("arm.*")
+    aarch64_regex = re.compile("aarch64.*")
+    power_regex = re.compile("ppc64.*")
+
+    system, node, release, version, machine, processor = platform.uname()
+    if aarch32_regex.match(machine):
+        arch = AARCH32
+    elif aarch64_regex.match(machine):
+        arch = AARCH64
+    elif power_regex.match(machine):
+        arch = POWER
+    elif machine == 'x86_64':
```

Comment:
rather than sorting alphabetically, I would try and rank these by how common they are, to short-circuit early in the most common case (i.e. `x86_64`)?